autowebcompat/network.py

```diff
@@ -185,22 +187,26 @@ def create_inception_network(input_shape):
     return Model(input, x)
 
 
-def create_resnet50_network(input_shape):
-    base_model = ResNet50(input_shape=input_shape)
+def create_resnet50_network(input_shape, weights):
+    base_model = ResNet50(input_shape=input_shape, weights=weights)
     return Model(inputs=base_model.input, outputs=base_model.get_layer('flatten_1').output)
 
 
 def create(input_shape, network='vgglike', weights=None):
     assert network in SUPPORTED_NETWORKS, '%s is an invalid network' % network
+    if weights is not None:
+        assert network in SUPPORTED_NETWORKS_WITH_WEIGHTS, '%s does not have weights for %s ' % (network, weights)
+        assert input_shape == (224, 224, 3), 'shape must be (224, 224, 3)'
+
     network_func = globals()['create_%s_network' % network]
-    base_network = network_func(input_shape)
+    base_network = network_func(input_shape, weights)
 
     input_a = Input(shape=input_shape)
     input_b = Input(shape=input_shape)
 
     # Loading pretrained weights corresponding to the network used
-    if weights:
-        base_network.load_weights(weights)
+    # if weights:
```

Comment:
Why commented out?

Reply:
Because if we load weights over here we have to specify the location of weights. For imagenet keras automatically downloads them.

Reply:
We should support both downloading by Keras and allowing custom weights to be loaded

Reply:
Done!